### PR TITLE
{Doc} Track 2 Migration Guidance, revision

### PR DESCRIPTION
**Pull request by qwordy**
**Description**

{Doc} Track 2 Migration Guidance, revision

Adding examples from more PRs including
[Network PR #16350](https://github.com/Azure/azure-cli/pull/16350)
[Storage PR #15845](https://github.com/Azure/azure-cli/pull/15845)
[KeyVault PR #14150](https://github.com/Azure/azure-cli/pull/14150)
[AppConfig PR #16376](https://github.com/Azure/azure-cli/pull/16376)

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: doc/track_2_migration_guidance.md [hunk 1]
```diff
@@ -70,14 +77,23 @@ Track 2 SDK removes enum type and adopts string type instead. It loses the valid
...
 Examples:
 
 In VMSS `begin_update_instances`, a new type `VirtualMachineScaleSetVMInstanceRequiredIDs` is added.
 
 In DiskAccess `begin_create_or_update`, location and tags are moved to a nested structure `DiskAccess`, `disk_access = DiskAccess(location=location, tags=tags)`
 
+In Storage,
+```
+track1: client.check_name_availability(account_name) // account_name is string
+track2: account_name = StorageAccountCheckNameAvailabilityParameters(name=name) client.check_name_availability(account_name)
```
Comment (jiasli) [inline doc/track_2_migration_guidance.md:91]: These are not valid Python code.

**Comment by yonzhan** [comment]
@jsntcy for awareness.

File: doc/track_2_migration_guidance.md [hunk 2]
```diff
@@ -70,14 +77,23 @@ Track 2 SDK removes enum type and adopts string type instead. It loses the valid
...
+```
+
+In AppConfig,
+`id` cannot be passed directly to `regenerate_key method`. It needs to be wrapped in the new model `RegenerateKeyParameters`.
```
Comment (yungezz) [inline doc/track_2_migration_guidance.md:95]: same, link to PR diff will be helpful

File: doc/track_2_migration_guidance.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ This document summarizes typical issues and solutions when adopting Track 2 SDK
 Example PRs:
 1. [Compute PR #15750](https://github.com/Azure/azure-cli/pull/15750)
 2. [Network PR #16245](https://github.com/Azure/azure-cli/pull/16245)
+3. [Network PR #16350](https://github.com/Azure/azure-cli/pull/16350)
+4. [Storage PR #15845](https://github.com/Azure/azure-cli/pull/15845)
+5. [KeyVault PR #14150](https://github.com/Azure/azure-cli/pull/14150)
+6. [AppConfig PR #16376](https://github.com/Azure/azure-cli/pull/16376)
```
Comment (qwordy) [inline doc/track_2_migration_guidance.md:13]:
```suggestion
6. [AppConfig PR #16376](https://github.com/Azure/azure-cli/pull/16376)
7. [AppService PR #17146](https://github.com/Azure/azure-cli/pull/17146)
```

File: doc/track_2_migration_guidance.md (hunk 1 above)
Comment (qwordy) [reply doc/track_2_migration_guidance.md]: Done

File: doc/track_2_migration_guidance.md (hunk 2 above)
Comment (qwordy) [reply doc/track_2_migration_guidance.md]: Done

**Review by jiasli — APPROVED** [review]

**Pull request merged**